From 6a75b19abbaf4e6fe83ee720ac95eebcf16e5051 Mon Sep 17 00:00:00 2001
From: Dong Aisheng <aisheng.dong@nxp.com>
Date: Tue, 5 Sep 2017 16:31:50 +0800
Subject: [PATCH 3307/5242] MLK-17491-2 serial: fsl_lpuart: improve the baud
 setting algorithm

commit  d5c512048fb2cb98a5567e98645c7168556fbfd5 from
https://source.codeaurora.org/external/imx/linux-imx.git

If "baud_diff == 0", it means we already found the exact matching baud
rate and no need try looping the left possible baud rates anymore.

So in this patch, we break out immediately once we find the right baud
rate to avoid the left meaningless loops.

Acked-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Dong Aisheng <aisheng.dong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/tty/serial/fsl_lpuart.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/tty/serial/fsl_lpuart.c b/drivers/tty/serial/fsl_lpuart.c
index c25b9f2..3907e72 100644
--- a/drivers/tty/serial/fsl_lpuart.c
+++ b/drivers/tty/serial/fsl_lpuart.c
@@ -1817,6 +1817,9 @@ static void lpuart32_shutdown(struct uart_port *port)
 			baud_diff = tmp_diff;
 			osr = tmp_osr;
 			sbr = tmp_sbr;
+
+			if (!baud_diff)
+				break;
 		}
 	}
 
-- 
1.7.9.5

